lib/SIL/Utils/MemoryLocations.cpp

```diff
@@ -173,6 +173,21 @@ void MemoryLocations::analyzeLocations(SILFunction *function) {
           }
         }
       }
+      if (auto *BAI = dyn_cast<BeginApplyInst>(&I)) {
+        auto yieldedValues = BAI->getYieldedValues();
+        for (auto index : indices(yieldedValues)) {
+          auto fnType = BAI->getSubstCalleeType();
+          SILArgumentConvention argConv(
+              fnType->getYields()[index].getConvention());
+          if (argConv.isIndirectConvention()) {
+            // These are not SIL addresses under -enable-sil-opaque-values
+            if (!argConv.isInoutConvention() &&
+                !function->getConventions().useLoweredAddresses())
```

Comment:
You should never need to check `useLoweredAddresses()` when querying the argument convention. For code like this that cares about the SIL convention, you're supposed to use the SIL-level convention API, like `isSILIndirect`

Reply:
Fixed.